### PR TITLE
experimental: local development server

### PR DESCRIPTION
**Pull request by glommer**
Developers using environments like cloudflare or other edge function providers cannot use the filesystem easily for local testing. And the ones with a file system usually don't necessarily know about the file: path.

We have been toying with the idea of spawning a local sqld server, but it has some disadvantages, mainly it is a big (30MB) Rust binary that needs to be downloaded separately.

We don't need any of the replication bits from sqld, so it may be easier to just spawn a webserver and use a local libsql or sqlite file and proxy the requests.

We'd have to have a hrana-server implementation in go, but that doesn't sound too hard.

A bit harder, is making sure that all the responses and error messages are the same as sqld would generate, and keeping that in sync.

This PR adds a very basic wrapper that shows how the devex could be. I didn't worry about returning any valid responses yet, just execute a statement, print it, and done. A first step!

### CODE REVIEW

**Comment by glommer** [comment]
This now supports almost all of http, except for named parameters.

Final binary is still 13MB and no extra dependencies 🎉

**Comment by glommer** [comment]
Ultimately I think if we merge this as-is, it is a bit of a bad DX, since it will not work with hrana.
But from the http PoV this seems ready to go and work great for me.

If we plug hrana this can be a great slim development server.

**Comment by glommer** [comment]
I added a warning saying what is not expected to work, so maybe good to go?

Screen Shot 2023-04-26 at 11 27 01 PM

File: internal/cmd/dev.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+			}{
+				Message: e.Error(),
+			},
+		},
+	}
+}
+
+var devCmd = &cobra.Command{
+	Use:               "dev",
+	Short:             "starts a development server for Turso",
```
Comment (haaawk) [inline internal/cmd/dev.go:106]: Nit: s/starts a development server/starts a local development server/

File: internal/cmd/dev.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		}
+
+		db, err := sql.Open("sqlite3", devFile)
+		if err != nil {
+			fmt.Println("Failed to connect to SQLite database:", err)
+			return err
+		}
+		defer db.Close()
+
+		r.POST("/", func(c *gin.Context) {
```
Comment (haaawk) [inline internal/cmd/dev.go:144]: shouldn't this be /queries?

Comment (haaawk) [reply internal/cmd/dev.go:144]: https://github.com/honzasp/sqld/blob/hrana2/docs/http_api.md

**Review by haaawk — APPROVED** [review]

**Comment by penberg** [comment]
@glommer The local server is experimental so you might as well merge it. If people keep tripping over lack of Hrana, can prioritize the work.

File: internal/cmd/dev.go (hunk 2 above)
Comment (glommer) [reply internal/cmd/dev.go:144]: I don't think so, at least this works now with http as is.

**Pull request merged**